osu.Game/Graphics/Containers/Markdown/OsuMarkdownContainer.cs

```diff
@@ -21,6 +23,12 @@ public OsuMarkdownContainer()
             LineSpacing = 21;
         }
 
+        [BackgroundDependencyLoader]
+        private void load(IAPIProvider api)
+        {
+            DocumentUrl = api.WebsiteRootUrl;
```

Comment:
This is a bit awkward, as it's being set after the base `MarkdownContainer` does this initial content parse, causing a second invalidation (because the `Text` is set earlier than the BDL run in the follow-up implementation https://github.com/ppy/osu/pull/12948/files#diff-d366f0bc690ce73e1c96348ae4b4928d038138b51c41d7556fb513637e4f8291R25).

A quick solution would be to move this into `CreateChildDependencies` to ensure it has precedence. If you go down this path then it definitely needs an inline comment explaining why it is done in this way.

Reply:
Sorry I'm not sure how to do this with `CreateChildDependencies`.

Could you please give a little example?

Reply:
Have attached a suggested change.

Reply:
Thanks!